### PR TITLE
Mjosten.constants.statusbar.powerup

### PR DESCRIPTION
**Pull request by mjosten**
Changes:
- Decompose DroidEntity file to DroidBasic and DroidUtil
- Added basic status bar that "listens" to Zerlin health and force fields
- Created abstract powerups base which is planned to be modularized with first class functions.
- Created constants file which contains all the constants of the game.

### CODE REVIEW

**Comment by mjosten** [comment]
♫ make it

**Comment by JoshuaAtherton** [comment]
- [ ] © nice job
- [ ] I think it looks good

File: public/js/HeroEntities.js [hunk 1]
```diff
@@ -634,8 +591,8 @@ class Lightsaber extends Entity {
 
 		this.prevBladeCollar = this.bladeCollar;
 		this.prevBladeTip = this.bladeTip;
-		this.bladeCollar = { x: collarXrotated * Z_SCALE + this.x, y: collarYrotated * Z_SCALE + this.y };
-		this.bladeTip = { x: tipXrotated * Z_SCALE + this.x, y: tipYrotated * Z_SCALE + this.y };
+		this.bladeCollar = { x: collarXrotated * zc.Z_SCALE + this.x, y: collarYrotated * zc.Z_SCALE + this.y };
+		this.bladeTip = { x: tipXrotated * zc.Z_SCALE + this.x, y: tipYrotated * zc.Z_SCALE + this.y };
 	}
```
Comment (steveng9) [inline public/js/HeroEntities.js]: This is excellent!

File: public/js/SceneManager.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
+/*
+Zerlin
+TCSS 491 - Computational Worlds
+Joshua Atherton, Michael Josten, Steven Golob
+*/
+
```
Comment (steveng9) [inline public/js/SceneManager.js]: Nice catch!

File: public/js/StatusBar.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+*
+* draw the background lighter bar the full length of the bar, this bar does not change.
+* draw a darker foreground on top of the lighter bar, this bar length is 
+* equal to (maxLength / maxSize) / currentSize
+*/
+class AbstractStatusBar extends Entity {
+    constructor(game, x, y, hasCriticalState) {
+        super(game, x, y, 0, 0);
+        this.displayText = sbc.STATUS_BAR_DISPLAY_TEXT;
+        this.hasCriticalState = hasCriticalState; //does the bar flash when it gets low.
```
Comment (steveng9) [inline public/js/StatusBar.js]: Don't you think that this could be automatic (don't bother making it configurable?)

Comment (steveng9) [reply public/js/StatusBar.js]: I guess unless we may only want the health bar to have a critical point and not the force bar, which is probably what you were thinking.

File: public/js/StatusBar.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+/*
+* This class is the health status bar.
+* This bar will keep track of zerlins health and display it for the
+* player to see the a visual representation of the current health.
+*/ 
+class HealthStatusBar extends AbstractStatusBar {
+    constructor(game, x, y) {
+        super(game, x, y, sbc.HEALTH_BAR_HAS_CRITICAL_STATE);
+        this.foregroundColor = "rgba(255, 0, 0, 1)"; //red
+        this.backgroundColor = "rgba(255, 126, 126, 1)"; //light red
```
Comment (steveng9) [inline public/js/StatusBar.js]: What do you think about setting a text color here to be more readable for against the bar color?

**Comment by steveng9** [comment]
Looks awesome Michael.

File: public/js/CollisionManager.js [hunk 5]
```diff
@@ -257,7 +260,7 @@ class CollisionManager {
 												 zerlinBox.x, zerlinBox.y, zerlinBox.width, zerlinBox.height);
 					if (zerlinCollision.collides) {
 						this.game.beams[i].isSizzling = true;
-						this.game.Zerlin.hits += this.game.clockTick * BEAM_HP_PER_SECOND;
+						this.game.Zerlin.hits += this.game.clockTick * dbConst.BEAM_HP_PER_SECOND;
```
Comment (steveng9) [inline public/js/CollisionManager.js]: don't forget to subtract HP here, and make the status bar continuous instead of only supporting integers

**Pull request merged**